api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirVisitDaoImpl.java

```diff
@@ -9,14 +9,51 @@
  */
 package org.openmrs.module.fhir2.api.dao.impl;
 
+import static org.hibernate.criterion.Restrictions.eq;
+import static org.hl7.fhir.r4.model.Encounter.SP_DATE;
+
+import javax.annotation.Nonnull;
+
+import java.util.Optional;
+
+import ca.uhn.fhir.rest.param.DateRangeParam;
+import ca.uhn.fhir.rest.param.ReferenceAndListParam;
+import ca.uhn.fhir.rest.param.TokenAndListParam;
 import lombok.AccessLevel;
 import lombok.Setter;
+import org.hibernate.Criteria;
 import org.openmrs.Visit;
 import org.openmrs.module.fhir2.api.dao.FhirVisitDao;
 import org.springframework.stereotype.Component;
 
 @Component
 @Setter(AccessLevel.PACKAGE)
-public class FhirVisitDaoImpl extends BaseFhirDao<Visit> implements FhirVisitDao {
+public class FhirVisitDaoImpl extends BaseEncounterDao<Visit> implements FhirVisitDao {
+	
+	@Override
+	protected void handleDate(Criteria criteria, DateRangeParam dateRangeParam) {
+		handleDateRange("startDatetime", dateRangeParam).ifPresent(criteria::add);
+	}
```

Comment:
Here we probably need to think of some slightly more complex logic. Encounters (and most OMRS objects) only have a single date and time, but visits actually have a start date and an end date. But maybe we can leave things as-is for now and make that into an enhancement.

Reply:
Yeah, I felt same while implementing it. Should I remove it or leave it for now?

Reply:
Let's leave it for now. I'll open another issue to remind myself to spec out how this should work.